src/components/Schedule/TableView.tsx

```diff
@@ -9,202 +10,81 @@ import {
   TableContainer,
 } from "@chakra-ui/react";
 
-export const TableView = () => {
+interface TableProps {
+  data: CourseSection[];
+}
+
+export const TableView = (props: TableProps) => {
+  const { data } = props;
+
+  var tableData = [
+    {
+      course: "",
+      schedule_time: "",
+      days: "",
+      term: "",
+      prof: "",
+      section: "",
+      start_end: "",
+      capacity: "",
+    },
+  ];
+
+  const populateTable = (courses: CourseSection[]) => {
+    tableData.pop();
+
+    for (var i = 0; i < courses.length; i++) {
+      var d = "";
+      var len = courses[i].meetingTimes.length;
+      for (var y = 0; y < len; y++) {
+        d = d + courses[i].meetingTimes[y].day.slice(0, 3) + " ";
+      }
+      tableData.push({
+        course: courses[i].CourseID.subject + " " + courses[i].CourseID.code,
+        schedule_time:
+          courses[i].meetingTimes[0].startTime +
+          " / " +
+          courses[i].meetingTimes[0].endTime,
+        days: d,
+        term: courses[i].CourseID.term,
+        prof: courses[i].professors[0].username,
+        section: (i + 1).toString(),
+        start_end: courses[i].startDate + " / " + courses[i].endDate,
+        capacity: courses[i].capacity.toString(),
+      });
+    }
+  };
```

Comment:
This works for now and it's not a blocker but in the future it would be much more efficient to just roll this logic into the `map` function on the original data. By doing it this way you are creating an additional unnecessary array `tableData`

Also if you're interested you could look into modernizing your code to ES6. For example in modern javascript there is basically no scenario where it's preferable to be using `var` instead of `let`/`const`. You can also replace all of your manual `for` loops with array functions such as `map` or `forEach`.

Reply:
Thank you jayden! def a good idea